solvebio/resource/dataset_snapshot_task.py

```diff
@@ -0,0 +1,54 @@
+from .apiresource import ListableAPIResource
+from .apiresource import CreateableAPIResource
+from .solveobject import convert_to_solve_object
+from .task import Task
+
+import time
+
+
+class DatasetSnapshotTask(CreateableAPIResource, ListableAPIResource):
+    """
+    DatasetSnapshotTask represents the task to snapshot and archive a dataset
+    """
+    RESOURCE_VERSION = 2
+
+    LIST_FIELDS = (
+        ('id', 'ID'),
+        ('dataset_id', 'Dataset'),
+        ('vault_id', 'Vault'),
+        ('status', 'Status'),
+        ('created_at', 'Created'),
+    )
+
+    @property
+    def dataset(self):
+        response = self._client.get(self['dataset']['url'], {})
+        return convert_to_solve_object(response, client=self._client)
+
+    def follow(self, loop=True, sleep_seconds=Task.SLEEP_WAIT_DEFAULT):
```

Comment:
Is there some way to abstract these follow() methods? maybe with some kind of ActivityResource or FollowableResource or something?

Reply:
yes, it could be added. each Task would need to define a custom "progress" function though. Also some tasks (imports/migrations) have commits that need to be followed

Reply:
Not a big deal, can always refactor later.